src/components/page/blocks/BlocksColumn.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Blocks from './index.jsx';
+
+function BlocksColumn({
+  getCol,
+  blocksGroups,
+  defaultLayout,
+  blockShared,
+  blockRows,
+}) {
+  const getLayout = layout => {
+    if (layout) {
+      const row = layout.row || defaultLayout.row;
+      const col = layout.col || defaultLayout.col;
+
+      return { row, col };
+    }
+
+    return defaultLayout;
+  };
+
+  const isInLayout = (layout, getRow) => {
+    const { row, col } = getLayout(layout);
+
+    return getCol === col && getRow === row;
+  };
+
+  const getRowBlockGroups = getRow =>
+    blocksGroups
+      .reduce((result, blockGroup) => {
+        if (blockGroup.blocks) {
+          result.push({
+            ...blockGroup,
+            blocks: blockGroup.blocks.filter(block => {
+              return isInLayout(block.layout, getRow);
+            }),
+          });
+        }
+
+        return result;
+      }, [])
+      .filter(blockGroup => blockGroup.blocks.length > 0);
+
+  return blockRows.map(getRow => {
+    const rowBlockGroups = getRowBlockGroups(getRow);
+
+    return rowBlockGroups.map((rowBlockGroup, i) => {
+      const { type, blocks } = rowBlockGroup;
+      const key = `${type}-${getRow}-${getCol}-${i}`;
```

Comment:
While we're in this world, what do you think about renaming these `getCol` and `getRow` variable names to `col` and `row`?  I don't know if they maybe were functions at some point, but I think everywhere they're used now they're just like strings, and to me at least, every time I read `getBlah` I'm expecting it to be a function.  Could be another quick separate ticket.  Or maybe is overly fastidious and pedantic.